controllers/replication/secret_controller.go

```diff
@@ -224,15 +225,21 @@ func (s *SecretSyncController) Reconcile(ctx context.Context, req ctrl.Request)
 			} else {
 				namespace = target.Namespace
 			}
+			if remoteClient == localClient && target.Namespace == sec.Namespace {
+				//TODO: @Michael does this allow us to remove the UID check in requiresUpdate or any other logic?
+				// If if origin and source context-namespace are the same we want to bail immediately.
+				continue TargetSecrets
+			}
 			fetchedSecret := &corev1.Secret{}
-			if err = remoteClient.Get(ctx, types.NamespacedName{Name: sec.Name, Namespace: namespace}, fetchedSecret); err != nil {
+			if err = remoteClient.Get(ctx, types.NamespacedName{Name: fmt.Sprintf("%s%s", target.TargetPrefix, sec.Name), Namespace: namespace}, fetchedSecret); err != nil {
```

Comment:
suggestion: I think this prefix logic naming secrets should be in a function, or otherwise some change will forget it..

Reply:
Done as requested.